protocol/x/ratelimit/util/ibc.go

```diff
@@ -0,0 +1,191 @@
+package util
```

Comment:
Utilities mostly re-used from https://github.com/Stride-Labs/stride/blob/121f2ac5d2e5f8e406f89999410a49ea4277a552/x/ratelimit

Reply:
Can you separate these util functions into two files:
1. exact copy from Stride
2. methods that made some modifications

Reply:
Done - moved exact copies and tests in to file `parse_denom.go` and `parse_denom_test.go`